bundle-workflow/python/build.py

```diff
@@ -3,31 +3,57 @@
 
 import os
 import sys
+import subprocess
 import tempfile
-import urllib.request
-from lib.build_manifest import BuildManifest
-from lib.build_output import BuildOutput
+import yaml
+from manifests.input_manifest import InputManifest
+from build_workflow.build_recorder import BuildRecorder
+from build_workflow.builder import Builder
+from build_workflow.git_repository import GitRepository
 
 if (len(sys.argv) < 2):
     print("Build an OpenSearch Bundle")
     print("usage: build.sh /path/to/manifest")
     exit(1)
 
-with tempfile.TemporaryDirectory() as work_dir:
-    manifest = BuildManifest.from_file(sys.argv[1])
-    build = manifest.build
-    output = BuildOutput()
+component_scripts_path = os.path.join(os.path.dirname(os.path.abspath(__file__)), '../scripts/bundle-build/components')
+default_build_path = os.path.join(os.path.dirname(os.path.abspath(__file__)), '../scripts/bundle-build/standard-gradle-build/build.sh')
+
+def get_arch():
+    arch = subprocess.check_output(['uname', '-m']).decode().strip()
+    if arch == 'x86_64':
+        return 'x64'
+    elif arch == 'aarch64' or arch == 'arm64':
+        return  'arm64'
+    else:
+        raise ValueError(f'Unsupported architecture: {arch}')
 
-    print(f'Building {build.name} ({output.arch}) into {output.dest} ...')
+arch = get_arch()
+manifest = InputManifest.from_file(sys.argv[1])
+output_dir = os.path.join(os.getcwd(), 'artifacts')
+os.makedirs(output_dir, exist_ok = True)
+
+with tempfile.TemporaryDirectory() as work_dir:
+    print(f'Building in {work_dir}')
 
     os.chdir(work_dir)
 
+    build_recorder = BuildRecorder(output_dir, manifest.build.name, manifest.build.version, arch)
+
+    print(f'Building {manifest.build.name} ({arch}) into {output_dir}')
+
     for component in manifest.components:
-        print(f'=== Building {component.name} ...')
-        component.checkout()
-        component.build(build.version, output.arch)
-        component.export(output.dest)
-
-    manifest_path = os.path.join(output.dest, 'manifest.yml')
-    manifest.save_to(manifest_path)
-    print(f'Done.')
```

Comment:
nit - I liked having this as a confirmation of success, can we put it at the end?